ignition-server/controllers/machineconfigserver_ignitionprovider.go

```diff
@@ -129,26 +134,44 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
 			return false, nil
 		}
 
+		// Get  Machine config certs
+		var caCert []byte
+		var ok bool
+		machineConfigServerCert := manifests.MachineConfigServerCert(p.Namespace)
+		if err := p.Client.Get(ctx, client.ObjectKeyFromObject(machineConfigServerCert), machineConfigServerCert); err != nil {
+			return false, fmt.Errorf("failed to get machine config server secret: %w", err)
+		}
+		if caCert, ok = machineConfigServerCert.Data["ca.crt"]; !ok {
+			return false, fmt.Errorf("failed to get Certificate from mcs-crt: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		if err != nil {
+			return false, fmt.Errorf("failed to load cert: %w", err)
+		}
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: caCertPool,
+				},
+			},
+			Timeout: 60 * time.Second,
```

Comment:
Instead of a fixed 60 second timeout on the client, I think we should be setting up the request with the `Context` argument passed to `GetPayload()`

Reply:
I think we need two layers but also realize why 5 seconds was applicable (should be less than a polling cycle so will keep it at 5